packages/base/src/util/AriaLabelHelper.ts

```diff
@@ -57,7 +62,56 @@ const getAssociatedLabelForTexts = (el: HTMLElement) => {
 	return undefined;
 };
 
+const observeAssosiatedLabels = (el: UI5Element, callback: MutationCallback) => {
```

Comment:
HI @mindarelus
general remark, does the change reflect the labels referenced with the `accessibleNameRef` public API? 

Many input components support the API (and this is also part of the [issue](https://github.com/SAP/ui5-webcomponents/issues/6315) - by the way it needs to be referenced in the commit body), like in this example:

```html
<ui5-label id="enterNameLabel">Enter name: </ui5-label>
<ui5-input accessible-name-ref="enterNameLabel"></ui5-input>
```

Because at first glance, the "for" attribute is subject of the PR.